### PR TITLE
TOOLS: added new atomic flags to ucx_info

### PR DESCRIPTION
**Pull request by hoopoepg**

### CODE REVIEW

File: src/tools/info/tl_info.c [hunk 1]
```diff
@@ -41,14 +41,42 @@
...
 
+#define _PRINT_ATOMIC_POST(_name, _cap, _op, _suffix) \
+    if (((_cap).atomic32._op & UCS_BIT(UCT_ATOMIC_OP_##_name)) ||  \
+        ((_cap).atomic64._op & UCS_BIT(UCT_ATOMIC_OP_##_name))) {  \
+        char *s = strduplower(#_name); \
+        char amo[256] = "atomic_" _suffix; \
+        char *domain = ""; \
+        strcat(amo, s); \
+        free(s); \
+        if ((_cap).flags & UCT_IFACE_FLAG_ATOMIC_CPU) { \
```
Comment (yosefe) [inline src/tools/info/tl_info.c]: the domain can be printed once per iface and not per every atomic type

File: src/tools/info/tl_info.c [hunk 2]
```diff
@@ -41,14 +41,42 @@
...
         free(s); \
     }
 
+#define _PRINT_ATOMIC_POST(_name, _cap, _op, _suffix) \
+    if (((_cap).atomic32._op & UCS_BIT(UCT_ATOMIC_OP_##_name)) ||  \
+        ((_cap).atomic64._op & UCS_BIT(UCT_ATOMIC_OP_##_name))) {  \
+        char *s = strduplower(#_name); \
+        char amo[256] = "atomic_" _suffix; \
+        char *domain = ""; \
+        strcat(amo, s); \
```
Comment (yosefe) [inline src/tools/info/tl_info.c]: strncat

File: src/tools/info/tl_info.c [hunk 3]
```diff
@@ -41,14 +41,42 @@
...
+            printf("#         %12s: 32, 64 bit%s (deprecated)\n", s, domain); \
         } else { \
-            printf("#         %12s: %d bit%s\n", s, \
+            printf("#         %12s: %d bit%s (deprecated)\n", s, \
                    ((_cap_flags) & UCT_IFACE_FLAG_##_name##32) ? 32 : 64, domain); \
         } \
         free(s); \
     }
 
+#define _PRINT_ATOMIC_POST(_name, _cap, _op, _suffix) \
```
Comment (yosefe) [inline src/tools/info/tl_info.c]: can this be partially extracted to a function? to avoid long macro

File: src/tools/info/tl_info.c (hunk 1 above)
Comment (hoopoepg) [reply src/tools/info/tl_info.c]: print domain as separate capability line?

Comment (yosefe) [reply src/tools/info/tl_info.c]: yes, if at least one atomic flag is present

Comment (hoopoepg) [reply src/tools/info/tl_info.c]: done

File: src/tools/info/tl_info.c (hunk 3 above)
Comment (hoopoepg) [reply src/tools/info/tl_info.c]: updated

File: src/tools/info/tl_info.c (hunk 2 above)
Comment (hoopoepg) [reply src/tools/info/tl_info.c]: fixed

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6756/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6760/ for details (Mellanox internal link).

**Pull request merged**